### PR TITLE
[dhctl] feat(dhctl-for-commander): server refactoring

### PR DESCRIPTION
**Pull request by faroyam**
## Description

- Reduced code duplication in the gRPC server message handler
- Reduced code duplication in the log sender
- Refactored the graceful shutdown mechanism
- Added support for proper log output for multiple parallel instances of the dhctl server

## What is the expected result?

It is not expected to change behaviour of any dhctl operation not in commander-mode. 

```changes
section: dhctl
type: feature
summary: Reduces code duplication in the gRPC server message handler and log sender, refactors the graceful shutdown mechanism, and adds support for proper log output for multiple parallel instances of the dhctl server.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: dhctl/pkg/server/rpc/dhctl/check.go [hunk 1]
```diff
@@ -54,10 +51,15 @@ func (s *Service) Check(server pb.DHCTL_CheckServer) error {
 	doneCh := make(chan struct{})
 	internalErrCh := make(chan error)
 	receiveCh := make(chan *pb.CheckRequest)
-	sendCh := make(chan *pb.CheckResponse, 100)
+	sendCh := make(chan *pb.CheckResponse)
```
Comment (distorhead) [inline dhctl/pkg/server/rpc/dhctl/check.go:54]: There was channel with 100-capacity, and now it is 1-capactiy. Is it by intention or typo?

Comment (faroyam) [reply dhctl/pkg/server/rpc/dhctl/check.go:54]: The number 100 was added for debugging purposes a while ago, and I forgot to change it back afterward. It should indeed be 0-capacity as intended. As in another tasks. Sorry for the oversight!

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by faroyam** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10718084610 10718084610-1-con-1-27 /sys/deckhouse-oss/install:pr9096

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by faroyam** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10735222306 10735222306-1-con-1-27 /sys/deckhouse-oss/install:pr9096 redos@89.169.150.116

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/aws'...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10735222306 10735222306-1-con-1-27 /sys/deckhouse-oss/install:pr9096 redos@89.169.150.116

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 10735574972 10735574972-1-con-1-27 /sys/deckhouse-oss/install:pr9096 ec2-user@18.156.24.193

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10734809194 10734809194-1-con-1-27 /sys/deckhouse-oss/install:pr9096

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @faroyam. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by aleksey-su — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Pull request merged**